src/inc/sift-events/sift-events.php

```diff
@@ -565,12 +569,17 @@ public static function update_or_create_order( string $order_id, \WC_Order $orde
 		}
 
 		// Determine user and session context.
-		$user_id   = wp_get_current_user()->ID ?? null; // Check first for logged-in user.
+		$user_id   = get_current_user_id() ?? wp_get_current_user()->ID ?? null; // Check first for logged-in user.
 		$is_system = ! $create_order && str_starts_with( sanitize_title( wp_unslash( $_SERVER['HTTP_USER_AGENT'] ?? '' ) ), 'WordPress' ); // Check if this is an order update via system action.
+		$is_admin  = 1 === $user_id;
 
 		// Figure out if it should use the session ID if no logged-in user exists.
-		if ( ! $user_id ) {
-			$user_id = $is_system ? $order->get_user_id() : null; // Use order user ID only for system actions.
+		if ( ! $user_id && $is_system ) {
+			$user_id = $order->get_user_id() ?? null; // Use order user ID if is a system update
+		}
+
+		if ( ! $user_id || $is_admin ) {
+			$user_id = $order->get_user_id() ?? null; // Use order user ID if it isn't available otherwise
```

Comment:
I have the feeling that this whole block can be made more readible... I don't understand much of the logic behind it...
If $user_id is null when getting there, getting the id from the $order should be tried, even if it is not system or admin, no?

Reply:
It probably can. I didn't know the reasoning behind getting the user_id from the order _only_ when it's a system update. I think the more simple logic is just to try to get the user_id from the order if the user_id isn't available or if it's admin (this would remove the `$is_system` check).